packages/block-editor/src/components/inserter/index.js

```diff
@@ -105,10 +119,67 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
-
+		const {
+			hasInserterItems,
+			__experimentalHasOnlyOneAllowedBlockType,
+			__experimentalGetTheOnlyAllowedBlockType,
+		} = select( 'core/block-editor' );
+		const allowedBlock = getBlockType( __experimentalGetTheOnlyAllowedBlockType( rootClientId ) );
 		return {
 			hasItems: hasInserterItems( rootClientId ),
+			hasOnlyOneAllowedInserterItem: __experimentalHasOnlyOneAllowedBlockType( rootClientId ),
+			blockTitle: allowedBlock ? allowedBlock.title : '',
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps, { select } ) => {
+		return {
+			insertTheOnlyAllowedItem: () => {
+				const { rootClientId, clientId, isAppender, destinationRootClientId } = ownProps;
+				const {
+					__experimentalHasOnlyOneAllowedBlockType,
+					__experimentalGetTheOnlyAllowedBlockType,
+				} = select( 'core/block-editor' );
+
+				const hasOnlyOneAllowedInserterItem = __experimentalHasOnlyOneAllowedBlockType( rootClientId );
+
+				if ( ! hasOnlyOneAllowedInserterItem ) {
+					return false;
```

Comment:
Does the return value do anything? If not, I think we can drop it.
```suggestion
					return;
```